src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiControllerTest.java

```diff
@@ -0,0 +1,96 @@
+
+    List<LrdOrgInfoServiceResponse> lrdOrgInfoServiceResponse = new ArrayList<>();
+    private JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverterMock;
+    private UserInfo userInfoMock;
+    HttpServletRequest httpRequest = mock(HttpServletRequest.class);
+
+    @Before
+    public void setUp() throws Exception {
+
+        lrdServiceMock = mock(LrdService.class);
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void testRetrieveOrgServiceDetailsByServiceCode() {
+        final HttpStatus expectedHttpStatus = HttpStatus.OK;
+        String serviceCode = "AAA1";
+        String ccdCaseType = null;
+        when(lrdServiceMock.findByServiceCodeOrCcdCaseTypeOrDefault(any(),any())).thenReturn(lrdOrgInfoServiceResponse);
+        ResponseEntity<?> actual = lrdApiController
+            .retrieveOrgServiceDetailsByServiceCodeOrCcdCaseType(serviceCode,ccdCaseType);
+        assertThat(actual).isNotNull();
+        assertThat(actual.getStatusCode()).isEqualTo(expectedHttpStatus);
+        verify(lrdServiceMock, times(1)).findByServiceCodeOrCcdCaseTypeOrDefault(any(),any());
+    }
+
+    @Test
+    public void testRetrieveOrgServiceDetailsByCcdCaseType() {
+        final HttpStatus expectedHttpStatus = HttpStatus.OK;
+        String serviceCode = null;
+        String ccdCaseType = "ccdCaseType1";
+        when(lrdServiceMock.findByServiceCodeOrCcdCaseTypeOrDefault(any(),any())).thenReturn(lrdOrgInfoServiceResponse);
+        ResponseEntity<?> actual = lrdApiController
+            .retrieveOrgServiceDetailsByServiceCodeOrCcdCaseType(serviceCode,ccdCaseType);
+        assertThat(actual).isNotNull();
+        assertThat(actual.getStatusCode()).isEqualTo(expectedHttpStatus);
+        verify(lrdServiceMock, times(1)).findByServiceCodeOrCcdCaseTypeOrDefault(any(),any());
+    }
+
+    @Test
+    public void testRetrieveOrgServiceDetailsByDefaultRequestParamsNull() {
+        final HttpStatus expectedHttpStatus = HttpStatus.OK;
+        String serviceCode = null;
+        String ccdCaseType = null;
+        when(lrdServiceMock.findByServiceCodeOrCcdCaseTypeOrDefault(any(),any())).thenReturn(lrdOrgInfoServiceResponse);
+        ResponseEntity<?> actual = lrdApiController
+            .retrieveOrgServiceDetailsByServiceCodeOrCcdCaseType(serviceCode,ccdCaseType);
+        assertThat(actual).isNotNull();
+        assertThat(actual.getStatusCode()).isEqualTo(expectedHttpStatus);
+        verify(lrdServiceMock, times(1)).findByServiceCodeOrCcdCaseTypeOrDefault(any(),any());
+    }
+
+    @Test(expected = InvalidRequestException.class)
+    public void testRetrieveOrgServiceDetailsShouldThrowExceptionWhenBothParamValuesPresent() {
+        final HttpStatus expectedHttpStatus = HttpStatus.BAD_REQUEST;
+        String serviceCode = "AAA1";
+        String ccdCaseType = "ccdCaseType1";
+        when(lrdServiceMock.findByServiceCodeOrCcdCaseTypeOrDefault(any(),any())).thenReturn(lrdOrgInfoServiceResponse);
```

Comment:
a `verify()` is needed for this `when(lrdServiceMock.`

Reply:
when cond is not required here so deleted now.